packages/nimble-components/src/table-column/date-text/tests/types.spec.ts

```diff
@@ -0,0 +1,99 @@
+        // @ts-expect-error This expect will fail if the enum-like type is missing "as const"
+        const format: MonthFormat = 'hello';
+        expect(format).toEqual('hello');
+    });
+
+    it('DayFormat fails compile if assigning arbitrary string values', () => {
+        // @ts-expect-error This expect will fail if the enum-like type is missing "as const"
+        const format: DayFormat = 'hello';
+        expect(format).toEqual('hello');
+    });
+
+    it('HourFormat fails compile if assigning arbitrary string values', () => {
+        // @ts-expect-error This expect will fail if the enum-like type is missing "as const"
+        const format: HourFormat = 'hello';
+        expect(format).toEqual('hello');
+    });
+
+    it('MinuteFormat fails compile if assigning arbitrary string values', () => {
+        // @ts-expect-error This expect will fail if the enum-like type is missing "as const"
+        const format: MinuteFormat = 'hello';
+        expect(format).toEqual('hello');
+    });
+
+    it('SecondFormat fails compile if assigning arbitrary string values', () => {
+        // @ts-expect-error This expect will fail if the enum-like type is missing "as const"
+        const format: SecondFormat = 'hello';
+        expect(format).toEqual('hello');
+    });
+
+    it('TimeZoneFormat fails compile if assigning arbitrary string values', () => {
+        // @ts-expect-error This expect will fail if the enum-like type is missing "as const"
+        const format: TimeZoneFormat = 'hello';
+        expect(format).toEqual('hello');
+    });
+
+    it('FormatMatcherAlgorithm fails compile if assigning arbitrary string values', () => {
+        // @ts-expect-error This expect will fail if the enum-like type is missing "as const"
+        const format: FormatMatcherAlgorithm = 'hello';
+        expect(format).toEqual('hello');
+    });
+
+    it('DayPeriodFormat fails compile if assigning arbitrary string values', () => {
+        // @ts-expect-error This expect will fail if the enum-like type is missing "as const"
+        const format: DayPeriodFormat = 'hello';
+        expect(format).toEqual('hello');
+    });
+
+    it('DateStyle fails compile if assigning arbitrary string values', () => {
+        // @ts-expect-error This expect will fail if the enum-like type is missing "as const"
+        const format: DateStyle = 'hello';
+        expect(format).toEqual('hello');
+    });
+
+    it('TimeStyle fails compile if assigning arbitrary string values', () => {
+        // @ts-expect-error This expect will fail if the enum-like type is missing "as const"
+        const format: TimeStyle = 'hello';
+        expect(format).toEqual('hello');
+    });
+
+    it('HourCycle fails compile if assigning arbitrary string values', () => {
+        // @ts-expect-error This expect will fail if the enum-like type is missing "as const"
+        const format: HourCycle = 'hello';
```

Comment:
make sure the references in this file get updated, the expect error may mask that